pkg/resourcemapping/operator_resource_mapping.go

```diff
@@ -0,0 +1,340 @@
+	componentKey := updatedControllerObj.GetKind() + "/" + updatedControllerObj.GetName()
+	ormClient.cacheLock.Lock()
+	defer ormClient.cacheLock.Unlock()
+	operatorResourceSpec, exists := ormClient.operatorResourceSpecMap[operatorCRUID]
+	if !exists {
+		return fmt.Errorf("operatorResourceSpec not found in operatorResourceSpecMap for operatorCR %s", operatorCRUID)
+	}
+	ormTemplate, exists := operatorResourceSpec.ormTemplateMap[componentKey]
+	if !exists {
+		return fmt.Errorf("ormTemplate not found in ormTemplateMap for componentKey %s for operatorCR %s", componentKey, operatorCRUID)
+	}
+
+	operatorResKind := controllerOwnerReference.Kind
+	operatorResName := controllerOwnerReference.Name
+	operatorRes := operatorResKind + "/" + operatorResName
+
+	resourceNamespace := updatedControllerObj.GetNamespace()
+	dynResourceClient := ormClient.dynClient.Resource(operatorResourceSpec.operatorGVResource).Namespace(resourceNamespace)
+	operatorCR, err := dynResourceClient.Get(operatorResName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to get CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
+	}
+
+	glog.V(2).Infof("Updating CR %s for %s in namespace %s", operatorRes, componentKey, resourceNamespace)
+	updated := false
+	// Update based on each resourceMappingTemplate
+	for _, resourceMappingTemplate := range ormTemplate.resourceMappingTemplates {
+		// Set resourceMappingComponentName to resourceMappingTemplate so as to parse the srcPath and destPath based on text template
+		resourceMappingTemplate[resourceMappingComponentName] = ormTemplate.componentName
+		srcPath, destPath, err := ormClient.parseSrcAndDestPath(resourceMappingTemplate)
+		if err != nil {
+			return fmt.Errorf("failed to update CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
+		}
+
+		// Check if CR needs update from current resourceMappingTemplate. If the values are same under the same srcPath
+		// from origControllerObj and updatedControllerObj, it means resource value is not changed and no need to update
+		// corresponding destPath in CR.
+		newValue, needsUpdate, err := ormClient.needsUpdate(origControllerObj, updatedControllerObj, srcPath)
+		if err != nil {
+			return fmt.Errorf("failed to update CR %s for %s in namespace %s: %v", operatorRes, componentKey, resourceNamespace, err)
+		}
+		if !needsUpdate {
+			glog.V(4).Infof("No changes in controller %s/%s src path '%s'. Skip updating Operator CR %s.",
+				updatedControllerObj.GetKind(), updatedControllerObj.GetName(), srcPath, operatorRes)
+			continue
+		}
+		fields := strings.Split(destPath, ".")
+		if len(fields) < 2 {
+			return fmt.Errorf("failed to update %v to CR %s for %s in namespace %s: '%s' is invalid path", newValue, operatorRes, componentKey, resourceNamespace, destPath)
+		}
+		origCRValue, _, err := util.NestedField(operatorCR, resourceMappingDestPath, destPath)
```

Comment:
I could not understand the need of this function wrt to the `unstructured.NestedField()`. What am I missing?

Reply:
Also, please validate the bool return.

Reply:
When calling func `unstructured.NestedFieldCopy(obj, fields ...string)`, the input `fields` can only be a certain key of a map but doesn't support filtering by a certain field like `containers[?(@.name=="api")]`. So in order to find the value of such path like `.spec.template.spec.containers[?(@.name=="api")].resources`, I introduced this new `util.NestedField` func.

Again it's valid case either `bool` as true (found the value to be replaced) or false (not found value so needs to be added). And the same logic below will handle both cases

Reply:
Ack!